src/Lumi/Components2/Image.purs

```diff
@@ -0,0 +1,189 @@
+import Lumi.Styles.Slat hiding (_interactive,slat) as Styles.Slat
+import Lumi.Styles.Theme (LumiTheme(..), useTheme)
+import React.Basic.DOM as R
+import React.Basic.Emotion as E
+import React.Basic.Events (handler_)
+import React.Basic.Hooks (JSX)
+import React.Basic.Hooks as Hooks
+
+data Image = Image
+
+type ImageProps
+  = ( component :: Image
+    , src :: String
+    , placeholder :: Maybe JSX
+    )
+
+-- | An image has no size restrictions
+-- | will flex fill it's container
+image :: LumiComponent ImageProps
+image =
+  unsafePerformEffect do
+    lumiComponent "Image" defaults \props -> Hooks.do
+      theme <- useTheme
+      loaded /\ setLoaded <- Hooks.useState' false
+      pure
+        $ E.element R.div'
+            { children:
+                [ if String.null props.src
+                    then fromMaybe placeholderSvg props.placeholder
+                    else
+                      Box.column
+                      $ Styles.Box._flex
+                      $ Styles.Box._align Center
+                      $ Styles.Box._justify Center
+                      $$$
+                        [ Monoid.guard (not loaded) $ placeholderSvg
+                            -- @TODO confirming with design
+                            -- $ loader { style: R.css { width: "20px", height: "20px", borderWidth: "2px" }, testId: Nullable.toNullable Nothing }
+                        , E.element R.img'
+                            { src: props.src
+                            , className: ""
+                            , css: E.css { maxWidth: E.percent 100.0 }
+                            , onLoad: handler_ $ setLoaded true
+                            }
+                        ]
+                ]
+            , className: props.className
+            , css: defaultImageStyle theme <> props.css theme
+            }
+    where
+      defaults =
+        { component: Image
+        , src: ""
+        , placeholder: Nothing
+        }
+
+      defaultImageStyle :: LumiTheme -> Style
+      defaultImageStyle theme@(LumiTheme { colors }) =
+          E.css
+            { boxSizing: E.str "border-box"
+            , overflow: E.str "hidden"
+            , display: E.str "flex"
+            , border: E.str "1px solid"
+            , borderColor: E.color colors.black4
+            }
+
+data Thumbnail = Thumbnail
+
+type ThumbnailProps
+  = ( component :: Thumbnail
+    , src :: String
+    , placeholder :: Maybe JSX
+    )
+
+-- | A thumbnail can support size restrictions
+thumbnail :: LumiComponent ThumbnailProps
+thumbnail =
+  unsafePerformEffect do
+    lumiComponent "Thumbnail" defaults \props -> Hooks.do
+      theme <- useTheme
+      pure
+        $ image
+        $ _ { src = props.src
+            , placeholder = props.placeholder
+            -- @TODO get this to compile
+            -- we should be setting the default square size on thumbnails
+            -- , css = theme # toCSS defaultSize <> props.css
```

Comment:
I believe we should always be setting the `thumbnail` to a default size and keep it at the square aspect ratio (using `image` doesn't have this restriction). I'm not sure why this line isn't compiling tho...